misc/toml-parser/src/main/java/org/ballerinalang/toml/util/SingletonStack.java

```diff
@@ -0,0 +1,63 @@
+/*
+ *  Copyright (c) 2018, WSO2 Inc. (http://www.wso2.org) All Rights Reserved.
+ *
+ *  WSO2 Inc. licenses this file to you under the Apache License,
+ *  Version 2.0 (the "License"); you may not use this file except
+ *  in compliance with the License.
+ *  You may obtain a copy of the License at
+ *
+ *  http://www.apache.org/licenses/LICENSE-2.0
+ *
+ *  Unless required by applicable law or agreed to in writing,
+ *  software distributed under the License is distributed on an
+ *  "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ *  KIND, either express or implied.  See the License for the
+ *  specific language governing permissions and limitations
+ *  under the License.
+ */
+package org.ballerinalang.toml.util;
+
+/**
+ * This class can be used to handle a singleton content with one element.
+ *
+ * @param <T>
+ * @since 0.964
+ */
+public class SingletonStack<T> {
+    private T content;
+
+    /**
+     * Push object to the content.
+     *
+     * @param obj
```

Comment:
Shall we add a description  here?

Reply:
Removed all the custom listeners from the toml-parser